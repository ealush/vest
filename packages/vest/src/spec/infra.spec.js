const { execSync } = require('child_process');
const glob = require('glob');
const vest = require('..');
const packagePath = require('../../../../util/packagePath');
const { version } = require('../../package.json');

describe('Vest exports', () => {
  test('All vest exports exist', () => {
    expect(vest).toMatchSnapshot({
      VERSION: expect.any(String),
    });
    expect(vest.VERSION).toBe(version);
  });
});

describe('General scenario tests', () => {
  let v, test, create, enforce;

  beforeAll(() => {
    ({ test, create, enforce } = vest);
  });

  describe('Base case', () => {
    it('Should match snapshot', () => {
      v = create('eveniet-maxime-ea', () => {
        test(
          'sed-minima-adipisci',
          'Aliquam reprehenderit iure omnis assumenda eligendi enim id praesentium numquam.',
          () => {
            enforce(1).equals(2);
          }
        );

        test('non-rem-dolorem', () => {
          enforce(1).inside([1, 2, 3]);
        });
      });
      expect(v()).toMatchSnapshot();
    });
  });

  describe('Exclusion via `only`', () => {
    it('Should match snapshot', () => {
      v = create('inventore-quis-impedit', () => {
        vest.only('doloribus-enim-quisquam');

        test(
          'doloribus-enim-quisquam',
          'Ea quia saepe modi corrupti possimus similique expedita inventore.',
          () => {
            enforce(1).notEquals(2);
          }
        );

        test('autem', () => null);
        test('soluta', () => null);
      });
      expect(v()).toMatchSnapshot();
    });
  });

  describe('Exclusion via `skip`', () => {
    it('Should match snapshot', () => {
      v = create('corrupti-alias-autem', () => {
        vest.skip('doloribus-enim-quisquam');

        test(
          'doloribus-enim-quisquam',
          'Ea quia saepe modi corrupti possimus similique expedita inventore.',
          () => {
            enforce(1).notEquals(2);
          }
        );

        test('autem', 'Temporibus ex ex.', () => {
          vest.warn();
          return 1 === 2;
        });
      });
      expect(v()).toMatchSnapshot();
    });
  });

  describe('Tests with warnings', () => {
    it('Should match snapshot', () => {
      v = create('corrupti-alias-autem', () => {
        vest.skip('doloribus-enim-quisquam');

        test(
          'doloribus-enim-quisquam',
          'Ea quia saepe modi corrupti possimus similique expedita inventore.',
          () => {
            enforce(1).notEquals(2);
          }
        );

        test('autem', 'Temporibus ex ex.', () => {
          vest.warn();
          return 1 === 2;
        });
        test('autem', () => {
          vest.warn();
          enforce(1).gt(0);
        });
      });
      expect(v()).toMatchSnapshot();
    });
  });

  describe('Async tests', () => {
    it('Should match snapshot', () =>
      new Promise(done => {
        v = create('molestias-veritatis-deserunt', () => {
          test(
            'doloribus-enim-quisquam',
            'Fuga odit ut quidem autem dolores ipsam.',
            () => Promise.resolve()
          );
          test(
            'doloribus-enim-quisquam',
            'Fuga odit ut quidem autem dolores ipsam.',
            () => Promise.reject()
          );
        });
        setTimeout(() => {
          expect(v()).toMatchSnapshot();
          done();
        });
      }));
  });
});

describe('TypeScript Typings', () => {
  const declarationFiles = glob
    .sync('src/typings/*.d.ts', {
      cwd: packagePath('vest'),
      absolute: true,
      ignore: '**/spec/*',
    })
    .join(' ');

  it(`Should pass tsc validation`, () => {
    expect(() => {
      execSync(`node_modules/.bin/tsc ${declarationFiles}`);
    }).not.toThrow();
  });

  it('Should fail Typescript check with failing file', () => {
    expect(() => {
      execSync(
        `node_modules/.bin/tsc ${packagePath(
          'vest/src/spec/failing.d.ts',
          'vest.d.ts'
        )}`
      );
    }).toThrow();
  });
});
