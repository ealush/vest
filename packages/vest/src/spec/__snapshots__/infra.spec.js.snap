// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General scenario tests Async tests Should match snapshot 1`] = `
Object {
  "done": [Function],
  "errorCount": 0,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "molestias-veritatis-deserunt",
  "testCount": 2,
  "tests": Object {
    "doloribus-enim-quisquam": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`General scenario tests Base case Should match snapshot 1`] = `
Object {
  "done": [Function],
  "errorCount": 1,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "eveniet-maxime-ea",
  "testCount": 2,
  "tests": Object {
    "non-rem-dolorem": Object {
      "errorCount": 0,
      "testCount": 1,
      "warnCount": 0,
    },
    "sed-minima-adipisci": Object {
      "errorCount": 1,
      "errors": Array [
        "Aliquam reprehenderit iure omnis assumenda eligendi enim id praesentium numquam.",
      ],
      "testCount": 1,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`General scenario tests Exclusion via \`only\` Should match snapshot 1`] = `
Object {
  "done": [Function],
  "errorCount": 0,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "inventore-quis-impedit",
  "testCount": 1,
  "tests": Object {
    "doloribus-enim-quisquam": Object {
      "errorCount": 0,
      "testCount": 1,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;

exports[`General scenario tests Exclusion via \`skip\` Should match snapshot 1`] = `
Object {
  "done": [Function],
  "errorCount": 0,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "corrupti-alias-autem",
  "testCount": 1,
  "tests": Object {
    "autem": Object {
      "errorCount": 0,
      "testCount": 1,
      "warnCount": 1,
      "warnings": Array [
        "Temporibus ex ex.",
      ],
    },
  },
  "warnCount": 1,
}
`;

exports[`General scenario tests Tests with warnings Should match snapshot 1`] = `
Object {
  "done": [Function],
  "errorCount": 0,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "corrupti-alias-autem",
  "testCount": 2,
  "tests": Object {
    "autem": Object {
      "errorCount": 0,
      "testCount": 2,
      "warnCount": 1,
      "warnings": Array [
        "Temporibus ex ex.",
      ],
    },
  },
  "warnCount": 1,
}
`;

exports[`Vest exports All vest exports exist 1`] = `
Object {
  "VERSION": Any<String>,
  "create": [Function],
  "draft": [Function],
  "enforce": [Function],
  "group": [Function],
  "only": [Function],
  "runWithContext": [Function],
  "skip": [Function],
  "test": [Function],
  "validate": [Function],
  "warn": [Function],
}
`;
